stable_diffusion_videos/upsampling.py

```diff
@@ -83,15 +86,22 @@ def from_pretrained(cls, model_name_or_path="nateraw/real-esrgan"):
             file = hf_hub_download(model_name_or_path, "RealESRGAN_x4plus.pth")
         return cls(file)
 
-    def upsample_imagefolder(self, in_dir, out_dir, suffix="out", outfile_ext=".png"):
+    def upsample_imagefolder(self, in_dir, out_dir, suffix="out", outfile_ext=".png", recursive=False, force=False):
         in_dir, out_dir = Path(in_dir), Path(out_dir)
         if not in_dir.exists():
             raise FileNotFoundError(f"Provided input directory {in_dir} does not exist")
 
         out_dir.mkdir(exist_ok=True, parents=True)
 
-        image_paths = [x for x in in_dir.glob("*") if x.suffix.lower() in [".png", ".jpg", ".jpeg"]]
-        for image in image_paths:
+        generator = in_dir.rglob("*") if recursive else in_dir.glob("*")
+        image_paths = [x for x in generator if x.suffix.lower() in [".png", ".jpg", ".jpeg"]]
+        n_img = len(image_paths)
+        for i, image in enumerate(image_paths):
+            out_filepath = out_dir / (str(image.relative_to(in_dir).with_suffix('')) + suffix + outfile_ext)
```

Comment:
Little hard to read, but dissected it locally and I see why you had to do this. Maybe we think about writing this cleaner later (or not 😅 )